serializer/serializer.go

```diff
@@ -460,15 +460,24 @@ func (s *Serializer) WriteTime(timeToWrite time.Time, errProducer ErrProducer) *
 		return s
 	}
 
-	if timeToWrite.IsZero() {
-		if err := binary.Write(&s.buf, binary.LittleEndian, int64(0)); err != nil {
-			s.err = errProducer(err)
-		}
+	// Convert the int64 timestamp to uint64 by truncating.
+	var unixNano int64
+	// Times whose unix timestamp in seconds would be larger than what fits into a
+	// nanosecond-precision int64 timestamp will be truncated to the max value.
+	if timeToWrite.Unix() > MaxNanoTimestampInt64Seconds {
+		unixNano = math.MaxInt64
 	} else {
-		if err := binary.Write(&s.buf, binary.LittleEndian, timeToWrite.UnixNano()); err != nil {
-			s.err = errProducer(err)
+		unixNano = timeToWrite.UnixNano()
+		// Times before the Unix Epoch will be truncated to the Unix Epoch.
+		if unixNano < 0 {
+			unixNano = 0
```

Comment:
Could this bit be written as something like `lo.Max(0, lo.Min(math.MaxInt64, timeToWrite.UnixNano()))`? Shorter, but probably less readable, so no need to change this.